---
title: "Front.Immunol.2024"
author: "Sagar Utturkar"
date: "August 2021"
output:
  prettydoc::html_pretty:
    theme: cayman
    toc: true
    highlight: github
---
# Introduction:

## Publication: [PMID: 38384467](https://pubmed.ncbi.nlm.nih.gov/38384467/)

Zhang F, Huang B, Utturkar SM, Luo W, Cresswell G, Herr SA, Zheng S, Napoleon JV, Jiang R, Zhang B, Liu M, Lanman N, Srinivasarao M, Ratliff TL, Low PS. **Tumor-specific activation of folate receptor beta enables reprogramming of immune cells in the tumor microenvironment.** Front Immunol. 2024 Feb 7;15:1354735. doi: 10.3389/fimmu.2024.1354735. PMID: 38384467; PMCID: PMC10879311. 


```{r setwd, warning=FALSE, message=FALSE, echo=FALSE, tidy=TRUE}

setwd("C:/Users/sutturka/Documents/GitHub/publication-codeshare/Front.Immunol.2024")

```


# Data Processing:
## Raw data processing through cell-ranger:

Processed raw data through Cell-Ranger.

```{bash, eval = F}

cellranger count --id=G1-tumor \
--transcriptome=/reference/refdata-gex-mm10-2020-A \
--fastqs=/input/G1-tumor \
--sample=G1-tumor \
--localcores 32  \
--expect-cells=5000

```




## Import to Seurat:
```{r import_1, warning=FALSE, message=FALSE, echo=TRUE, tidy=TRUE, eval = F}

library(Seurat)
library(dplyr)
library(tidyverse)
library(knitr)
library(kableExtra)
library(ggplot2)
library(cowplot)
library(data.table)
library(dplyr)
library(RColorBrewer)

# import cell-ranger matrix
sample <- Read10X(data.dir = "/1_Cell_ranger/output/G1-tumor/outs/filtered_feature_bc_matrix")
sample <- CreateSeuratObject(counts = sample, project = "G1-tumor", min.cells = 3, min.features = 200)

```



# Quality Control:

Seurat allows to easily explore QC metrics and filter cells based on various quality criteria. A few commonly used QC metrics are given below:

1. Transcript (nCount) and gene (nFeature) abundance
	- Low-quality cells or empty droplets will often have very few genes/transcripts
	- Cell doublets or multiplets may exhibit an aberrantly high gene count
2. The percentage of reads that map to the mitochondrial genome
	- Low-quality / dying cells often exhibit extensive mitochondrial contamination
	- Mitochondrial QC metrics is calculated as percentage of counts originating from a set of mitochondrial genes (i.e. all genes starting with MT)
3. Ribosomal contents
	- Less than 50% ribosomal content is often preferred


```{r QC_1, warning=FALSE, message=FALSE, echo=TRUE, tidy=TRUE, eval = F}

VlnPlot(sample, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, pt.size = 0)

VlnPlot(sample, features = c('percent.mt','percent.ribo'), pt.size = 0)

FeatureScatter(object=sample, feature1='percent.mt', feature2='nCount_RNA', pt.size = 0.1, plot.cor = F)

```

![**Figure 1**](C:/Users/sutturka/Documents/GitHub/publication-codeshare/Front.Immunol.2024/data/Seurat_1.png)  

